first-party/jib-spring-boot-extension-gradle/src/main/java/com/google/cloud/tools/jib/gradle/extension/springboot/JibSpringBootExtension.java

```diff
@@ -0,0 +1,123 @@
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger.LogLevel;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.UnknownTaskException;
+import org.gradle.api.tasks.TaskProvider;
+import org.springframework.boot.gradle.tasks.bundling.BootJar;
+
+public class JibSpringBootExtension implements JibGradlePluginExtension<Void> {
+
+  @Override
+  public Optional<Class<Void>> getExtraConfigType() {
+    return Optional.empty();
+  }
+
+  @Override
+  public ContainerBuildPlan extendContainerBuildPlan(
+      ContainerBuildPlan buildPlan,
+      Map<String, String> properties,
+      Optional<Void> config,
+      GradleData gradleData,
+      ExtensionLogger logger)
+      throws JibPluginExtensionException {
+    logger.log(LogLevel.LIFECYCLE, "Running Jib Spring Boot extension");
+
+    if (!shouldExcludeDevtools(gradleData.getProject(), properties, logger)) {
+      logger.log(LogLevel.INFO, "Keeping spring-boot-devtools (if any)");
+      return buildPlan;
+    }
+    logger.log(LogLevel.INFO, "Removing spring-boot-devtools (if any)");
+
+    List<LayerObject> newLayers =
+        buildPlan
+            .getLayers()
+            .stream()
+            .map(JibSpringBootExtension::filterOutDevtools)
+            .collect(Collectors.toList());
+    return buildPlan.toBuilder().setLayers(newLayers).build();
+  }
+
+  @VisibleForTesting
+  static boolean isDevtoolsJar(File file) {
+    return file.getName().startsWith("spring-boot-devtools-") && file.getName().endsWith(".jar");
+  }
+
+  @VisibleForTesting
+  static boolean shouldExcludeDevtools(
+      Project project, Map<String, String> extensionProperties, ExtensionLogger logger) {
+    try {
+      TaskProvider<Task> taskProvider = project.getTasks().named("bootJar");
+      if (taskProvider.getOrNull() instanceof BootJar) {
+        BootJar bootJar = ((BootJar) taskProvider.get());
+
+        boolean useDeprecatedExcludeDevtools =
+            Boolean.valueOf(
+                extensionProperties.getOrDefault("useDeprecatedExcludeDevtoolsOption", "false"));
+        if (useDeprecatedExcludeDevtools) {
+          return bootJar.isExcludeDevtools();
+        }
+
+        // Since Spring 2.3.0, "excludeDevtools" is deprecated and has no effect. (Also its default
+        // is no longer "true".) The new guide to include devtools is to explicitly add
+        // "developmentOnly" to "bootJar.classpath".
+        // https://docs.spring.io/spring-boot/docs/current/gradle-plugin/reference/html/#packaging-executable-configuring-including-development-only-dependencies
+        boolean noDevtoolsJarOnClasspath =
```

Comment:
I think I would prefer the style
```
return bootJar.getClasspath().filter(JibSpringBootExtension::isDevtoolsJar).isEmpty();
```
but maybe you have a reason for this?

Reply:
Just for readability, from my perspective. The method name is `shouldExcludeDevtools`, and it may not immediately clear what this line is returning in relation to whether to exclude devtools.

Reply:
sgtm